### DISCUSSION TITLE
World Bound

### DISCUSSION THREAD

**Discussion by 0xF6** [bepu/bepuphysics2#222] [Q&A] [answered]
After object getting max value of position, thier has vector3(NaN, NaN, NaN) and get exception with message:
```
The change in tree cost is an invalid value, strongly implying the tree bounds have been corrupted by infinites or NaNs. 
If this happened in the broad phase's use of the tree, 
it's likely that there are invalid poses or velocities in the simulation, 
possibly as a result of bugged input state or constraint configuration. 
Try running the library with debug asserts 
enabled to narrow down where the NaNsplosion started.
```

So, im trying to collecting object abroad bounds world width after each `_sim.Timestep(delta);`

```csharp
const float Bound = 10_000f;

...

bool greatOrEq(float val) => float.IsNaN(val) || !float.IsFinite(val) || val >= Bound;
bool lessOrEq(float val) => float.IsNaN(val) || !float.IsFinite(val) || val <= -Bound;

var body = _sim.Bodies.GetBodyReference(obj.Handler);

var (x, y, z) = body.Pose.Position;

if (greatOrEq(x) || greatOrEq(x) || greatOrEq(x))
    DropObject(obj); 
else if (lessOrEq(x) || lessOrEq(y) || lessOrEq(z))
    DropObject(obj);
```
And drop fn:
```csharp
private void DropObject(NetworkObject net)
{
    _sim.Bodies.Remove(net.Handler);
    ...
}
```

So, i get randomly behaviour
1. get same exception
2. get segfault 🗿
3. get stack overflow

How to kill *sinful* objects? (yes, i may have an object with a furious acceleration, I would like to drop it)

**Comment by RossNordby** [comment, marked as answer]
Assuming it's arising from object state, given that you don't observe any obvious failures in the position, it's probably in the velocities somewhere. But there could be other problems afoot- for example, asynchronous writes to the simulation state while it's in the middle of a timestep will tend to make it explode in unpredictable ways (including this one). The fact that you're seeing segfaults somewhat increases the probability that there are Greater Problems floating around.

I'd recommend grabbing the source and running in debug mode; there are a ton of asserts that will *probably* narrow down where the bad values come from.

**Reply by 0xF6** [reply]
So, yeah, problem in speed of simulation, too fast.
